src/repr/src/adt/timestamp.rs

```diff
@@ -12,19 +12,31 @@
 use std::fmt::Display;
 use std::ops::Sub;
 
-use ::chrono::{DateTime, Datelike, Duration, NaiveDate, NaiveDateTime, NaiveTime, Utc};
+use ::chrono::{
+    DateTime, Datelike, Days, Duration, Months, NaiveDate, NaiveDateTime, NaiveTime, Utc,
+};
 use mz_ore::cast::CastFrom;
 use mz_proto::{RustType, TryFromProtoError};
 use once_cell::sync::Lazy;
 use serde::{Serialize, Serializer};
 use thiserror::Error;
 
+use crate::adt::interval::Interval;
 use crate::adt::numeric::DecimalLike;
 use crate::chrono::ProtoNaiveDateTime;
 use crate::Datum;
 
 include!(concat!(env!("OUT_DIR"), "/mz_repr.adt.date.rs"));
 
+const MONTHS_PER_YEAR: i64 = 12;
+const HOURS_PER_DAY: i64 = 24;
+const MINUTES_PER_HOUR: i64 = 60;
+const SECONDS_PER_MINUTE: i64 = 60;
+
+const NANOSECONDS_PER_HOUR: i64 = NANOSECONDS_PER_MINUTE * MINUTES_PER_HOUR;
+const NANOSECONDS_PER_MINUTE: i64 = NANOSECONDS_PER_SECOND * SECONDS_PER_MINUTE;
+const NANOSECONDS_PER_SECOND: i64 = 10i64.pow(9);
```

Comment:
Talked offline, I defined some new constants because they're ones used specifically for determining age, e.g. `NANOSECONDS_PER_HOUR`. I updated the PR though to redefine others, e.g. `HOURS_PER_DAY`, as `i64`s but using the values you linked here.